cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java

```diff
@@ -113,8 +113,17 @@ public void create(HttpRequest request, HttpResponder responder, @PathParam("nam
       }
 
       NamespaceConfig config = metadata.getConfig();
-      if (config != null && !Strings.isNullOrEmpty(config.getSchedulerQueueName())) {
-        builder.setSchedulerQueueName(config.getSchedulerQueueName());
+      if (config != null) {
+        if (!Strings.isNullOrEmpty(config.getSchedulerQueueName())) {
+          builder.setSchedulerQueueName(config.getSchedulerQueueName());
+        }
+        // both or none of the following two must be set
```

Comment:
do we enforce this constraint? if so, it seems like the if condition on line 124 should be nested inside the if condition on line 121?